satellite/metainfo/metainfo.go

```diff
@@ -1158,6 +1163,33 @@ func (endpoint *Endpoint) GetObject(ctx context.Context, req *pb.ObjectGetReques
 
 	if pointer.Remote != nil {
 		object.RedundancyScheme = pointer.Remote.Redundancy
+	} else if streamMeta.NumberOfSegments == 0 || streamMeta.NumberOfSegments > 1 {
+		// workaround
```

Comment:
if it's a workaround does that mean we have a plan for handling this in the future?

super nit but there are a few spelling errors in the comment chunk below-- here are some edits that might also help with clarity:
```
	// The new metainfo API redundancy scheme is on object level (not per segment).
                // Because of that, RS is always taken from the last segment.
                // The old implementation saves RS per segment, and in some cases
                // when the remote file's last segment is an inline segment, we end up
		// missing an RS scheme. This loop will search for RS in segments other than the last one.
```

Reply:
> if it's a workaround does that mean we have a plan for handling this in the future?

Maybe, I'm not sure. Maybe we can do migration for files without RS in last segment but it needs to be discussed. 

Thank you for fixing spelling errors :)